core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/WorkflowCompiler.scala

```diff
@@ -62,6 +62,23 @@ class WorkflowCompiler(val workflowInfo: WorkflowInfo, val context: WorkflowCont
     val outLinksImmutable: Map[OperatorIdentifier, Set[OperatorIdentifier]] =
       outLinksImmutableValue.toMap
 
+    val inLinksHighestOrdinal: mutable.Map[OperatorIdentifier, Int] =
+      mutable.Map().withDefault(_ => 0)
+    val inLinksOrdinalMap: mutable.Map[OperatorIdentifier, mutable.Map[OperatorIdentifier, Int]] =
+      mutable.Map().withDefault(_ => mutable.Map())
+    workflowInfo.links.foreach(link => {
+      val origin = OperatorIdentifier(this.context.workflowID, link.origin)
+      val dest = OperatorIdentifier(this.context.workflowID, link.destination)
+      val nextOrdinal = inLinksHighestOrdinal(dest)
+      inLinksHighestOrdinal.update(dest, nextOrdinal + 1)
```

Comment:
if `dest` is not in the `inLinksHighestOrdinal`, inLinksHighestOrdinal(dest) will be 0. So, if an operator has 3 inputs wouldn't the inputNum for them would be set as 1 for each. I think I am not understanding the logic correctly.

Reply:
// for first iput
inLinksHighestOrdinal(dest) // 0, default value
inLinksHighestOrdinal.update(dest, nextOrdinal + 1) // becomes 1
// for second input
inLinksHighestOrdinal(dest) // 1, set in above
inLinksHighestOrdinal.update(dest, nextOrdinal + 1) // becomes 2

Reply:
for first input
```
  val nextOrdinal = inLinksHighestOrdinal(dest) // nextOrdinal  is 0, default value
  inLinksHighestOrdinal.update(dest, nextOrdinal + 1) // in the map, dest corresponds to 1 now
```

for second input
```
  val nextOrdinal = inLinksHighestOrdinal(dest) // nextOrdinal is 1, set above
  inLinksHighestOrdinal.update(dest, nextOrdinal + 1) // in the map, dest corresponds to 2 now
```